### PR TITLE
[12.0][ADD] web_set_single_page_hidden

### PR DESCRIPTION
**Pull request by vvrossem**
[Task](https://gestion.coopiteasy.be/web#id=5270&view_type=form&model=project.task)

In 12.0, on a _PO_, _SO_, _WH/IN_, _WH/OUT_, or, more generally, anything that has _lines_ in its `form` view uses the `Pager` widget (i.e. `1-10/10`) to count and visualize the number of items.

A `Pager` has `@param {boolean} [options.single_page_hidden] (not) to display the pager if only one page`.
On `init` function, this `single_page_hidden` is set to `false`.
On `_render` function, the `Pager` is hidden `if (size === 0 || (this.options.single_page_hidden && this._singlePage()))`

`FieldX2Many`, an extension of `AbstractField` instantiates a `Pager` with `option` `single_page_hidden: true,` in its `_renderControlPanel` function. It's impossible to `require` this field since it has not been made available (i.e no `return { FieldX2Many: FieldX2Many}`) in Odoo.

Two relational fields - `FieldOne2Many` & `FieldMany2Many` - extends `FieldX2Many` and can be required, therefore inherited.
Their `start` function is overriden to set `single_page_hidden = false` on their `Pager` widget.

In [Javascript Reference](https://www.odoo.com/documentation/12.0/reference/javascript_reference.html#widget-lifecycle), you can read:
1. Widget Lifecycle
2. `start` function _must return a deferred to indicate when its work is done._

Before: (without `1-4/4`)
before

After: (with `1-4/4`)
after

### CODE REVIEW

File: web_set_single_page_hidden/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (houssine78) [inline web_set_single_page_hidden/__manifest__.py:4]: missing copyright header

File: web_set_single_page_hidden/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    "name": "web_set_single_page_hidden",
+    "summary": """
+    Set single_page_hidden to false on Pager widget options
+    """,
+    "author": "Coop IT Easy SCRLfs",
+    "license": "AGPL-3",
+    "version": "12.0.1.0.0",
+    "website": "www.coopiteasy.be",
```
Comment (houssine78) [inline web_set_single_page_hidden/__manifest__.py]: add https:

**Review by houssine78 — APPROVED** [review]

**Comment by houssine78** [comment]
@vvrossem I think that this module is a good candidate for OCA/web ! what do you think it's worth to take some "hours" to have it merged on OCA repository !

**Comment by vvrossem** [comment]
@houssine78 then I'll make this module "configurable" in `res.config.settings` :) thanks for the review

File: web_set_single_page_hidden/readme/DESCRIPTION.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+This modules modifies `FieldOne2Many` and `FieldMany2Many` widgets:
```
Comment (mclaeysb) [inline web_set_single_page_hidden/readme/DESCRIPTION.rst]: I'd add what the final goal is in layman's terms: to always visualise the line total count

File: web_set_single_page_hidden/static/src/js/relational_fields.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+odoo.define("web_set_single_page_hidden.relational_fields", function (require) {
```
Comment (mclaeysb) [inline web_set_single_page_hidden/static/src/js/relational_fields.js:1]: Learned a lot!

**Review by mclaeysb — COMMENTED** [review]
I don't know enough JS to approve but learned a lot and am glad we succeeded to build this tiny thing!
Great job and thanks for the detailed explanation at the start! 💪

**Comment by vvrossem** [comment]
closing in favor of [OCA's PR #1723](https://github.com/OCA/web/pull/1723/)
